src/CFG-based-system.typ

```diff
@@ -1 +1,141 @@
-= CFG Based System
\ No newline at end of file
+#import "./proof-tree.typ": *
+#import "./vars.typ": *
+= CFG Based System
+
+Basic rules for Uniqueness and Borrowing won't change here,
+but adapted to fit the CFG representation of Kotlin.
+
+The key gap between the CFG version and the type rules is at the function call.
+Type checking at function calls need adjustment so that it can be done in control flow order.
+The support for nested function call is alse missing is previous typing rules.
+
+== Formalization
+#grid(
+  columns: (auto, auto),
+  column-gutter: 2em,
+  row-gutter: 1em,
+  [*CFG nodes not covered in Grammar*],[*Context*],
+  frame-box(
+    $
+      af &::= unique | shared \
+      beta &::= dot | borrowed \
+      p &::= x | p.f \
+      "TA" &::= "$"i := p, "TA for temp assignment" \
+      "F" &::= f("$"1: af beta, ..., "$"n: af beta): af \
+      "Join" &::= j \
+    $
+  ),
+  frame-box(
+    $
+      alpha &::= unique | shared | top \
+      beta &::= dot | borrowed \
+      Delta &::= dot | p : alpha beta, Delta \
+      Gamma &::= dot | i : Delta, Gamma
```

Comment:
Unclear what `Delta` means here; from our conversation this should be a list of path, path permission pairs

Reply:
while `Delta` itself is a list of path to alpha beta pair, as defined in the previous line

Reply:
Ohh, I see, because you're working with lists... right.  It would be good to mention in the text why you're setting things up this way.